### PR TITLE
duckdb: add recipe

### PR DESCRIPTION
**Pull request by toge**
Specify library name and version:  **duckdb/0.5.1**

---

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/duckdb/all/conanfile.py [hunk 1]
```diff
@@ -0,0 +1,197 @@
...
+        rmdir(self, os.path.join(self.package_folder, "lib", "cmake"))
+        rmdir(self, os.path.join(self.package_folder, "cmake"))
+
+    def package_info(self):
+        if self.options.shared:
+            self.cpp_info.libs = ["duckdb"]
+        else:
+            self.cpp_info.libs = [
+                "duckdb_static",
+                "duckdb_fmt",
```
Comment (prince-chrismc) [inline recipes/duckdb/all/conanfile.py:160]: Does this vendor packagesm

Comment (toge) [reply recipes/duckdb/all/conanfile.py:160]: Yes.
These packages have been modified to have duckdb own namespaces. 
like this : https://github.com/duckdb/duckdb/blob/master/third_party/fmt/include/fmt/core.h#L163-L166

Comment (prince-chrismc) [reply recipes/duckdb/all/conanfile.py:160]: Ughs that's like death for us lol okay.

File: recipes/duckdb/all/conanfile.py [hunk 2]
```diff
@@ -0,0 +1,197 @@
...
+        if self.options.shared:
+            try:
+                del self.options.fPIC
+            except Exception:
+                pass
+
+    def layout(self):
+        cmake_layout(self, src_folder="src")
+
+    def requirements(self):
```
Comment (prince-chrismc) [inline recipes/duckdb/all/conanfile.py:81]:
```suggestion
    def requirements(self):
        # FIXME: duckdb vendors a bunch of deps by modify the source code to have their own namespace 
```

**Review by prince-chrismc — COMMENTED** [review]
Let's just add a note so we know

**Review by prince-chrismc — DISMISSED** [review]

**Review by prince-chrismc — APPROVED** [review]

**Review by uilianries — APPROVED** [review]

**Pull request merged**